### ISSUE TITLE
Auth login for API shouldn't search in LDAP if the user is authenticated in GLPI DB

### ISSUE THREAD

**Issue by bazingavp** [glpi-project/glpi#15262]
### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

10.0.8

### Bug description

When I request the initSession from API, with app_token and user_token, it generates a PHP error (see error details in next section).
The issue doesn't occur when requested with login and password.

### Relevant log output

```shell
glpiphplog.WARNING:   *** PHP User Warning (512): LDAP search with base DN `(|(<LDAP_FILTER>)(<LDAP_FILTER>))` and filter `(& (samaccountname=<LOGIN>) (&(objectClass=user)(objectCategory=person)))` failed
error: Invalid DN syntax (34)
extended error: 0000208F: NameErr: DSID-03100232, problem 2006 (BAD_NAME), data 8349, best match of:
	'(|(<LDAP_FILTER>)(<LDAP_FILTER>))'

err string: 0000208F: NameErr: DSID-03100232, problem 2006 (BAD_NAME), data 8349, best match of:
	'(|(<LDAP_FILTER>)(<LDAP_FILTER>))'
 in \src\AuthLDAP.php at line 3541
  Backtrace :
  src\AuthLDAP.php:3541                              trigger_error()
  src\Auth.php:834                                   AuthLDAP::searchUserDn()
  src\Api\API.php:301                                Auth->login()
  src\Api\APIRest.php:150                            Glpi\Api\API->initSession()
  apirest.php:57                                     Glpi\Api\APIRest->call()
  public\index.php:82                                require()
```

### Page URL

apirest.php/initSession

### Steps To reproduce

User with GLPI DB authentication, with generated user_token and app_token
Call apirest.php/initSession

### Your GLPI setup information

GLPI 10.0.8 ( => D:\XXX\glpi)
Installation mode: TARBALL
Current language:fr_FR

Operating system: Windows NT SERVERNAME 10.0 build 20348 (Windows Server 2022) AMD64
PHP 8.2.8 cgi-fcgi (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, bcmath, calendar, cgi-fcgi, ctype, curl, date,
	dom, exif, fileinfo, filter, gd, hash, iconv, imap, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd, odbc, openssl, pcre,
	random, readline, session, sqlsrv, standard, tokenizer, xml, xmlreader, xmlwriter, zlib)
Setup: max_execution_time="600" memory_limit="512M" post_max_size="99M" safe_mode="" session.save_handler="files"
	upload_max_filesize="10M" 
Software: Microsoft-IIS/10.0
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.79
Server Software: mariadb.org binary distribution
	Server Version: 10.4.12-MariaDB
	Server SQL Mode: STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: GLPIUSER@localhost/glpi
	Host info: localhost via TCP/IP
	
PHP version (8.2.8) is supported.PHP version (8.2.8) is supported.
Sessions configuration is OK.Sessions configuration is OK.
Allocated memory is sufficient.Allocated memory is sufficient.
mysqli extension is installed.mysqli extension is installed.
Following extensions are installed: dom, fileinfo, filter, libxml, json, simplexml, xmlreader, xmlwriter.Following extensions are installed: dom, fileinfo, filter, libxml, json, simplexml, xmlreader, xmlwriter.
curl extension is installed.curl extension is installed.
gd extension is installed.gd extension is installed.
intl extension is installed.intl extension is installed.
zlib extension is installed.zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (10.4.12) is supported.Database engine version (10.4.12) is supported.
No files from previous GLPI version detected.No files from previous GLPI version detected.
The log file has been created successfully.The log file has been created successfully.
Write access to D:\XXX\glpi/files/_cache has been validated. Write access to D:\XXX\glpi/config has been validated. Write access to D:\XXX\glpi/files/_cron has been validated. Write access to D:\XXX\glpi/files has been validated. Write access to D:\XXX\glpi/files/_dumps has been validated. Write access to D:\XXX\glpi/files/_graphs has been validated. Write access to D:\XXX\glpi/files/_lock has been validated. Write access to D:\XXX\glpi/files/_pictures has been validated. Write access to D:\XXX\glpi/files/_plugins has been validated. Write access to D:\XXX\glpi/files/_rss has been validated. Write access to D:\XXX\glpi/files/_sessions has been validated. Write access to D:\XXX\glpi/files/_tmp has been validated. Write access to D:\XXX\glpi/files/_uploads has been validated.Write access to D:\XXX\glpi/files/_cache has been validated.
Write access to D:\XXX\glpi/config has been validated.
Write access to D:\XXX\glpi/files/_cron has been validated.
Write access to D:\XXX\glpi/files has been validated.
Write access to D:\XXX\glpi/files/_dumps has been validated.
Write access to D:\XXX\glpi/files/_graphs has been validated.
Write access to D:\XXX\glpi/files/_lock has been validated.
Write access to D:\XXX\glpi/files/_pictures has been validated.
Write access to D:\XXX\glpi/files/_plugins has been validated.
Write access to D:\XXX\glpi/files/_rss has been validated.
Write access to D:\XXX\glpi/files/_sessions has been validated.
Write access to D:\XXX\glpi/files/_tmp has been validated.
Write access to D:\XXX\glpi/files/_uploads has been validated.

Web server root directory configuration seems safe.Web server root directory configuration seems safe.
Sessions configuration is secured.Sessions configuration is secured.
OS and PHP are relying on 64 bits integers.OS and PHP are relying on 64 bits integers.
exif extension is installed.exif extension is installed.
ldap extension is installed.ldap extension is installed.
openssl extension is installed.openssl extension is installed.
Following extensions are installed: Phar. Following extensions are not present: bz2, zip.Following extensions are installed: Phar.
Following extensions are not present: bz2, zip.
Zend OPcache extension is installed.Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring. Following extensions are not present: sodium.Following extensions are installed: ctype, iconv, mbstring.
Following extensions are not present: sodium.
Write access to D:\XXX\glpi/marketplace has been validated.Write access to D:\XXX\glpi/marketplace has been validated.
Timezones seems loaded in database.Timezones seems loaded in database.

### Anything else?

As a temporary fix, I added this condition in the Auth.php file, line 811, before the loop on the ldap_servers.
However I think there must be a better way to fix it.

/*** Temporary fix ***
if (isset($this->user->fields["authtype"])
    && $this->user->fields["authtype"] !== 1) {
/*** End temporary fix ***
          foreach ($ldapservers as $ldap_method) {
               .......
          }
/*** Temporary fix ***
}
/*** End temporary fix ***

**Comment by cedric-anne** [comment]
Hi,

I confirm this case is not handled correctly.

**Comment by cedric-anne** [comment]
Fixed by #16224.
